### PR TITLE
wdi5 tests

### PR DESCRIPTION
**Pull request by diya-dhan**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# [submodule "mock-srv"]
+# 	path = ./mock-srv
+# 	url = https://github.tools.sap/I584814/cloud-extension-s4hana-business-process
+# 	branch = mock
+[submodule "mock-srv"]
+	path = mock-srv
+	url = ./
+	branch = mock
```
Comment (navinkrishnan) [inline .gitmodules:8]: submodule should point to -> https://github.com/SAP-samples/cloud-extension-ecc-business-process/tree/mock

File: mission/create-software-tests/create-ui-tests.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+The library has a set of 'Actions' and 'Assertions' defined for different Fiori templates using which the browser can be controlled. This minimizes the need to explicitly define the controls and views of a page making the test scripts more UI5 friendly.
+
+    NOTE : For this mission wdi5^1.5.0 will beused and thus wdio^7 will also be used**
+
+<br>
+
+## Environment Configuration
+
+As mentioned above, the wdio.conf.js file inside app/BusinessPartners holds all the configuration details in the config object.
```
Comment (navinkrishnan) [inline mission/create-software-tests/create-ui-tests.md:49]:
```suggestion
As mentioned above, all the relevant configuration details are added in the `wdio.conf.js` file located within the `app/BusinessPartners` directory.
```

File: mission/create-software-tests/create-ui-tests.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+    NOTE : For this mission wdi5^1.5.0 will beused and thus wdio^7 will also be used**
+
+<br>
+
+## Environment Configuration
+
+As mentioned above, the wdio.conf.js file inside app/BusinessPartners holds all the configuration details in the config object. 
+
+The package.json, in the same folder, has the corresponding dependencies which supports the configurations mentioned. The entire set up was done using the WDIO test runner's @wdio/cli.
```
Comment (navinkrishnan) [inline mission/create-software-tests/create-ui-tests.md:51]:
```suggestion
In the same folder, the package.json file contains the necessary dependencies that support the mentioned configurations. The entire setup was accomplished using the WDIO test runner's @wdio/cli.
```

File: mission/create-software-tests/create-ui-tests.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+| **Capabilities** : Chrome browser         | chromedriver                                  | Executes the tests on the chrome browser, the chromedriver dependency is needed to create the browser sessions |
+| **Framework** : Mocha          | @wdio/mocha-framework                         | Allows us to write tests using the framework, the dependency acts like an adapter package  |
+| **Reporters** : Spec             | @wdio/spec-reporter                           | To generate reports in spec style |
+| **Services** : Chromedriver       | wdio-chromedriver-service                     | Enables communications with the chrome browser drivers |
+| **Services** : UI5               | wdio-ui5-service                              | Allows the usage of wdi5 |
+
+Other details mentioned in the configuration file includes : the path of the test files, browser timeouts, hooks to be executed etc.
+
+If you would like to try writing your own tests, [this would be the place to start](https://v7.webdriver.io/docs/gettingstarted/). You can follow the steps mentioned here to set up your testing environment.
```
Comment (navinkrishnan) [inline mission/create-software-tests/create-ui-tests.md:65]:
```suggestion
If you're interested in writing your own tests, [this link](https://v7.webdriver.io/docs/gettingstarted/) is a great starting point. You can follow the steps provided there to set up your testing environment.
```

File: mission/create-software-tests/create-ui-tests.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+If you would like to try writing your own tests, [this would be the place to start](https://v7.webdriver.io/docs/gettingstarted/). You can follow the steps mentioned here to set up your testing environment. 
+
+<br>
+
+## Understand the Test Cases
+
+Now lets understand the structure of the tests in brief.
+
+1. The test.spec.js file insider tests/ui/specs, includes:
```
Comment (navinkrishnan) [inline mission/create-software-tests/create-ui-tests.md:73]:
```suggestion
1. The `test.spec.js` file insider `tests/ui/specs`, includes:
```

File: mission/create-software-tests/create-ui-tests.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+
+        ![config](./images/ui-test-2.png)
+
+    - the test suites, which are essentially the family of tests to be executed
+    
+2. In order to test the working of the application entirely, the testing begins with creating **test data**. In this case this means creating **mock business partners**. 
+
+    Since this is only a testing environment, actual S4 systems will not be used but rather a 'mock-server', which is a CAP application that will enable us to create entities and emit events. 
+    
+    This is mentioned as a **Git Submodule** in this repository as can be seen in the .gitmodules file in the root folder. The test data is created using axios calls to this mock server. (These calls can be found in the tests/ui/services/bpApi.js file)
```
Comment (navinkrishnan) [inline mission/create-software-tests/create-ui-tests.md:85]:
```suggestion
This is referenced as a **Git Submodule** within this repository, as indicated in the **`.gitmodules`** file located in the root folder. The test data is generated through axios calls made to this mock server (you can find these calls in the `tests/ui/services/bpApi.js` file). 
```

File: mission/create-software-tests/create-ui-tests.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
+# Run UI Tests
+
+## Introduction
+
+Another set of tests that can be executed are the UI tests, where the application's interface is assessed on the basis of performance and functionality. The main objective would be to test the features that a user would interact with. This is done by mimicing the user actions and automating them on the browser with help of some automation framework. These tests would ensure optimal user experience by minimizing glitches and slow load times and validate whether the desired functionalities can be carried out.
```
[the remark this answers is not in the document]
Comment (diya-dhan) [reply mission/create-software-tests/create-ui-tests.md:5]: change implemented

File: mission/create-software-tests/create-ui-tests.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+### Prequisites:
+1. Nodejs environment
+2. Running UI5 application
+3. Chrome browser 
+
+<br>
+
+## An introduction to the Webdriver.IO (WDIO) framework
+
+For this mission the [Webdriver.IO framework](https://webdriver.io/), also known as WDIO, is used to perform the browser automation. It is rooted in node.js and is built on top of the selenium webdriver API using javascript. This means that node.js functions as the runtime environment, responsible for executing the test scripts.
```
[the remark this answers is not in the document]
Comment (diya-dhan) [reply mission/create-software-tests/create-ui-tests.md:20]: done

**Pull request merged**